ICDS - Update Aggregation and View for Dashboard v2

sheelio opened the pull request:
@emord 

This does a few things:
- Adds new columns to the existing aggregate tables
- Updates the views for those aggregate tables
- Adds a new aggregate table (for daily tracking)

One open TODO is purging any daily tables > 15 days old.  Not sure the best way to do that.

Commit:
GroupDim working

Commit:
add user_ids to GroupStaging

Commit:
Add removed_user_ids

Commit:
Basic insertion into user group dim working

Commit:
move setup of users to setUpClass

Commit:
Handle deleted users

Commit:
update migrations

Commit:
Add new indicators to aggregate tables

Commit:
Add daily table

Commit:
Update views for tables

Commit:
Add migration

Commit:
Update child monthly functions

Commit:
remove categories from view

Commit:
Add CCS record cases

Commit:
Fix fully immunized eligible

Commit:
Aggregate new ccs indicators

Commit:
Fix bug in awc daily view

Commit:
Add infant weighing scale indicator

Commit:
Add infant weighing scale indicator

snopoke committed:
remove selected and set is_advanced

Commit:
Add num launched indicator

Commit:
Add num launched indicator

Commit:
Add household indicator

Commit:
Add household indicator

Commit:
Add child and ccs all

Commit:
Add child and ccs all

Commit:
Add infant weighing scale

Commit:
Add infant weighing scale

Commit:
Add person indicators

Commit:
Add person indicators

Commit:
Add functions to migration

Commit:
Add functions to migration

Commit:
Add variable declarations

Commit:
Add variable declarations

Commit:
Fix view bug

Commit:
Fix view bug

calellowitz committed:
changing build comment

orangejenny committed:
Ability to add reg forms

orangejenny committed:
Allo deletion of reg forms

orangejenny committed:
Merge branch 'master' into v2-reg-form-changes

orangejenny committed:
v2 diffs

Commit:
Fix insert into child monthly

Commit:
Fix insert into child monthly

Commit:
Better handle deleted field

dannyroberts committed:
fix experience editing case import comment while in progress

before it would kick you out of the box when the upload list refreshed

dannyroberts committed:
add log_repeater_error_in_datadog

and use it to handle fire_for_record exceptions well

orangejenny committed:
Allow rearranging reg forms

emord committed:
Updated raven (sentry client).

sravfeyn committed:
fix docupdate conflict

esoergel committed:
Show SQLLocations in the raw_doc view

This would've actually been useful for a long time, didn't think to do it earlier.

Commit:
Add ebf no info recorded

Commit:
Add ebf no info recorded

orangejenny committed:
Tweak display conditions

sravfeyn committed:
refetch only if needed

sravfeyn committed:
fix import

orangejenny committed:
App builder v2: Fix releases page blank screening

sravfeyn committed:
careful

Commit:
Delete then Repopulate UserGroupDim

Commit:
Add agg_awc calculation

Commit:
Add agg_awc calculation

Commit:
add domain to group

dannyroberts committed:
refactor long js equality expressions in case_importer/js/import_history.js

dannyroberts committed:
simplify case import history refresh logic

Commit:
Add new states to user indicator task

Commit:
Fix daily aggregation bugs

Commit:
Fix daily aggregation bugs

millerdev committed:
Refactor liveness classification code into closure

dannyroberts committed:
fix stickler comments

Commit:
Add daily table to aggregation task

Commit:
Add daily table to aggregation task

millerdev committed:
Clean up prev_ids/next_ids

millerdev committed:
Move more code into related indices timing context

millerdev committed:
Use named constant to indicate circular reference

millerdev committed:
Remove silly logging micro-optimization

Verified: logging is disabled at DEBUG level on production envs, so `log.isEnabledFor(DEBUG)` will return false and string formatting for logging of large case id sets will be skipped.

esoergel committed:
group_by doesn't seem to apply for form reports

Getting a validation error here: https://manage.dimagi.com/default.asp?257168
Introduced in https://github.com/dimagi/commcare-hq/pull/16602

millerdev committed:
Add timing context for open root cases

orangejenny committed:
Merge pull request #16842 from dimagi/tweak-display-conditions

Tweak display conditions

orangejenny committed:
Fix error on remote apps in custom properties domains

Commit:
Add UP site code

Commit:
Stickler

Commit:
Stickler

millerdev committed:
Add toggle for livequery case sync algorithm

esoergel committed:
Merge pull request #16838 from dimagi/je/upgrade-sentry

Updated raven (sentry client).

calellowitz committed:
fixing orphan analysis command for new requirements

calellowitz committed:
updating relevant locations

benrudolph committed:
Merge pull request #16839 from dimagi/br/user-grou-dim

User Group Dim 2

calellowitz committed:
[ci skip] lint

millerdev committed:
get_payload -> do_livequery

esoergel committed:
Prepopulate group_by widget

millerdev committed:
Async livequery

orangejenny committed:
Merge pull request #16846 from dimagi/remote-500-custom-props

Fix error on remote apps in custom properties domains

dannyroberts committed:
Merge pull request #16836 from dimagi/edit-import-history-comment-while-in-progress

fix experience editing case import comment while in progress

dannyroberts committed:
Merge pull request #16837 from dimagi/log_repeater_error_in_datadog

add log_repeater_error_in_datadog

millerdev committed:
lint

orangejenny committed:
Merge pull request #16843 from dimagi/v2-releases-page-loading

App builder v2: Fix releases page blank screening

emord committed:
Added more exceptions that happen regularly on ICDS.

millerdev committed:
Merge pull request #16848 from dimagi/ce/combining-orphan-data

Ce/combining orphan data

emord committed:
Removed working config ids from the record.

emord committed:
Added check to make sure config_ids should be changed.

emord committed:
Fixed import.

Commit:
Cache XML text instead of Element object

Commit:
Update tests to match parsing

calellowitz committed:
Merge pull request #16767 from dimagi/ce/async-build-users

adding username parameter to async builds

calellowitz committed:
using server time and including hh names

esoergel committed:
Store report_class in redis and check permissions on that

esoergel committed:
Also check for view all reports permission

Commit:
Merge pull request #16847 from dimagi/gc/new-states

Add new states to user indicator task

esoergel committed:
lint

millerdev committed:
Merge pull request #16849 from dimagi/dm/livequery-flag

Release livequery case sync algorithm behind flag

sheelio commented:
@emord 
It looks like @kkrampa has added a migration to specify views that are already present in the aggregation scripts (they're on master, not in this PR).  Is that going to create issues with the views that are being manually created?

sravfeyn committed:
Merge pull request #16840 from dimagi/sr-demo-mode

fix docupdate conflict

sravfeyn committed:
Merge pull request #16816 from dimagi/mk-remove-csrf-assert

remove csrf assert notification

Commit:
Merge pull request #16851 from dimagi/gc/251017

Cache XML text instead of Element object

emord committed:
Merge pull request #16853 from dimagi/es/export-permissions

Store report_class in redis for permissions check

emord committed:
Merge pull request #16850 from dimagi/je/robust-async

More robust async indicators

esoergel committed:
Bad copy/paste [ci skip]

esoergel committed:
Merge pull request #16845 from dimagi/es/report-builder

Fix validation error on form-based report builder configs

emord commented:
@sheelio do you mean these? https://github.com/dimagi/commcare-hq/blob/master/custom/icds_reports/migrations/0016_aggawcmonthly_aggccsrecordmonthly_aggchildhealthmonthly_aggdailyusageview_aggthrmonthly_awclocation_.py

It looks like managed = False so it should be fine https://docs.djangoproject.com/en/1.11/ref/models/options/#managed

emord committed:
Merge pull request #16852 from dimagi/ce/no-more-orphans

using server time and including hh names

emord commented on custom/icds_reports/migrations/0016_add_v2_indicators.py:
```diff
@@ -0,0 +1,21 @@
+migrator = RawSQLMigration(('custom', 'icds_reports', 'migrations', 'sql_templates'))
+
+
+class Migration(migrations.Migration):
```
You'll probably need to rebase branch this onto master and rename this migration and change the dependency

emord reviewed (changes requested):
I think I was wrong actually. Since you're add columns you'll probably want to change the model definitions that he wrote. and make another migration

dannyroberts committed:
allow revoking both is_staff and is_superuser in SuperuserManagement

previously if you selected nothing
the form would fail (shallowly) with a validation error

emord committed:
update staging.yaml

[ci skip]

emord committed:
update staging.yaml

[ci skip]

dannyroberts committed:
move dhis2 urls into dhis2.urls

Commit:
archive old tables, create new ones

proteusvacuum committed:
Remove Location from shell_plus imports

esoergel committed:
Merge pull request #16857 from dimagi/shell-plus-locations

Remove Location from shell_plus imports

NoahCarnahan committed:
Merge pull request #16855 from dimagi/revoke-superuser

allow revoking both is_staff and is_superuser in SuperuserManagement

Commit:
Add management command to aggregate sms counts by state and indicator

Commit:
LiveQuery toggle for formplayer

Commit:
Merge branch 'master' into wsp/livequery-formplayer

Commit:
use existing flag

Commit:
Revert "use existing flag"

This reverts commit 6a81480213452a1c4b6b6d2a5bff4b5e60271b7a.

Commit:
Merge pull request #16858 from dimagi/gc/icds-sms-counts

ICDS: Add management command to aggregate sms counts by state and indicator

nickpell committed:
Merge pull request #16856 from dimagi/np/tb/new_tables

archive old tables, create new ones

biyeun committed:
enable phone number but no AB test

calellowitz committed:
Merge pull request #16860 from dimagi/bmb/registration-phone-enable

enable phone number but no AB test

orangejenny committed:
Move base_template.html's report creation into standard_hq_report.js

orangejenny committed:
django out of report_loader.html

orangejenny committed:
centralize popover initialization

dannyroberts committed:
move dhis2 views into dhis2.views

orangejenny committed:
Generate asyncHQReport in standard_hq_report.js

dannyroberts committed:
move dhis2 templates and js into dhis2

orangejenny committed:
don't define js_options twice in global admin reports

dannyroberts committed:
move corehq.apps.repeaters to corehq.motech.repeaters

dannyroberts committed:
move corehq.apps.dhis2 to corehq.motech.dhis2

orangejenny committed:
Admin reports: only include standard_hq_report.js once

dannyroberts committed:
fix broken ugettext import

orangejenny committed:
drop report_loader.html

Commit:
rename toggle

orangejenny committed:
comments

dannyroberts committed:
move repeater views into repeaters app

orangejenny committed:
Move event handlers into standard_hq_report.js

Commit:
[ci-skip]

proteusvacuum committed:
Add models for app user data permissions

proteusvacuum committed:
Only sync reports for webapps with permissions

proteusvacuum committed:
Add UI to configure user data permisisons

esoergel committed:
Set value as an html attr

dannyroberts committed:
remove unused imports

proteusvacuum committed:
Add toggle

orangejenny committed:
Get inline js all the way out of standard/base_template.html

Commit:
Merge pull request #16859 from dimagi/wsp/livequery-formplayer

LiveQuery for Formplayer

esoergel committed:
update staging.yaml

[ci skip]

proteusvacuum committed:
Always enable the ENIKSHAY toggle for enikshay

orangejenny committed:
lint & return async report

Commit:
fix JSON encoding

esoergel committed:
Merge pull request #16867 from dimagi/wsp/livequery-formplayer

Fix JSON encoding

proteusvacuum committed:
Merge pull request #16865 from dimagi/fr/always-enable-enikshay

Always enable the ENIKSHAY toggle for enikshay

proteusvacuum committed:
Merge pull request #16864 from dimagi/fr/webapps-permissions

Webapps Permissions by user data

orangejenny committed:
Django out of project_health_dashboard.html

orangejenny committed:
split out project_health_dashboard.js

Commit:
update staging.yaml

[ci skip]

Commit:
Fixed enikshay location filter

orangejenny committed:
lint

orangejenny committed:
bit more lint

proteusvacuum committed:
Merge pull request #16861 from dimagi/isolate-dhis2

Isolate dhis2

emord committed:
Merge pull request #16862 from dimagi/admin-report-dedup-script

Global admin reports: only include report_loader.html once

NoahCarnahan committed:
Merge pull request #16872 from lwyszomi/257243-case

Fixed enikshay location filter

Commit:
Added multi_field_dynamic_choice_list to allowed filter types

Commit:
Add site code to view

Commit:
Add site code to view

NoahCarnahan committed:
Add date filter to referral report

proteusvacuum committed:
Add View Web Apps to roles

orangejenny committed:
Allow putting other forms before reg form

Commit:
Don't fail if instance is not found

NoahCarnahan committed:
[refactor] fix up indentation

snopoke committed:
Merge pull request #16787 from dimagi/sk/meta

don't wrap metadata

snopoke committed:
Merge pull request #16817 from dimagi/sk/export-edited-on

add edited_on system property for form exports

biyeun committed:
Merge pull request #16745 from dimagi/reload-on-rearrange

App manager v2: reload page on rearrange forms/modules

esoergel committed:
Handle inconsistent datatypes in reports

https://sentry.io/dimagi/commcarehq/issues/298931135/
It looks like these values are all either '1' or '{"sort_key": 0, "html": 0}'
It's supposed to represent a boolean, so a better fix would be to always format
that as either 0 or 1, but I'm not confident in my ability to test out
potentially wide-reaching changes, so I'd rather just put in this quick fix.

NoahCarnahan committed:
Hide case type if property filter does not match any of its properties

dannyroberts committed:
Merge branch 'master' into motech-package

proteusvacuum committed:
Revert "Add toggle"

This reverts commit 0fc8bd54a1d237a98bc41343779fd229122e9c45.

proteusvacuum committed:
Revert "Add UI to configure user data permisisons"

This reverts commit 205e4e70dc1ed4bfdb556c20336affa3ab3204ef.

proteusvacuum committed:
Revert "Only sync reports for webapps with permissions"

This reverts commit 4ec4ade44993abc39ee84da108fc199c64ad605a.

proteusvacuum committed:
Revert "Add models for app user data permissions"

This reverts commit 6f45d78e8eb0c2dd725510657ce5c3985361a869.

NoahCarnahan committed:
Merge pull request #16874 from dimagi/nc/referral-report-clean-3

enikshay: Add date filter to static referral report

proteusvacuum committed:
Hide webapps from views

proteusvacuum committed:
Add toggle

proteusvacuum committed:
Don't sync reports for unavailable apps

proteusvacuum committed:
Add device_id to fixture_generator util

proteusvacuum committed:
Merge pull request #16863 from dimagi/motech-package

Motech package

proteusvacuum committed:
don't assume a wrapped app

orangejenny committed:
Can't create new case export in domains that have a remote app

orangejenny committed:
Merge pull request #16866 from dimagi/externalize-js-report-base-template

Externalize JavaScript: reports standard base template

orangejenny committed:
Merge pull request #16831 from dimagi/v2-reg-form-changes

App manager v2: reg form changes

orangejenny committed:
Kill download_*_export.html

orangejenny committed:
JS error in reports home

emord committed:
Merge pull request #16875 from dimagi/gc/instance-not-found

Don't fail if schedule instance is not found

orangejenny committed:
Merge pull request #16881 from dimagi/js-error-reports-home

JS error in reports home

Commit:
Merge pull request #16880 from dimagi/kill-dl-foo-export-html

Kill download_*_export.html

esoergel committed:
Add "edited_on" field to Forms API

esoergel committed:
Merge pull request #16876 from dimagi/es/intrahealth

Handle inconsistent datatypes in reports

esoergel committed:
Merge pull request #16841 from dimagi/es/raw-doc-sqllocations

Show SQLLocations in the raw_doc view

proteusvacuum committed:
Merge pull request #16878 from dimagi/fr/webapps-role

Webapps restriction by role

millerdev committed:
Merge pull request #16869 from dimagi/externalize-js-project-health-dashboard

Externalize js: Project health dashboard

NoahCarnahan committed:
Merge pull request #16873 from lwyszomi/added-multi-field-to-allowed

Added multi_field_dynamic_choice_list to allowed filter types

orangejenny committed:
Merge pull request #16879 from dimagi/remote-apps-exports

Can't create new case export in domains that have a remote app

orangejenny committed:
Fix reports-inline-js blocks in custom reports

proteusvacuum committed:
WebUser has no "role"

orangejenny committed:
Merge pull request #16885 from dimagi/fix-custom-reports

Fix reports-inline-js blocks in custom reports

orangejenny committed:
Merge pull request #16886 from dimagi/fr/get_role

WebUser has no "role"

Commit:
Add management command to tag older icds sms with indicator metadata

orangejenny committed:
Revert externalizing js for reports

proteusvacuum committed:
Merge pull request #16888 from dimagi/revert-externalize-js-report-base-template

Revert externalizing js for reports

biyeun committed:
Merge pull request #16889 from dimagi/es/ongoing-builder-woes

Set value as an html attr

biyeun committed:
fix issue with build_comment being undefined.

biyeun committed:
stickler

biyeun committed:
Merge pull request #16890 from dimagi/bmb/build-comment-error

fix issue with build_comment being undefined.

orangejenny committed:
Revert "Revert externalizing js for reports"

orangejenny committed:
Don't duplicate reportConfigEditor logic

snopoke committed:
Merge pull request #16883 from dimagi/es/edited_on

Add "edited_on" field to Forms API

orangejenny committed:
Merge pull request #16891 from dimagi/revert-16888-revert-externalize-js-report-base-template

Revert 16888 revert externalize js report base template

kaapstorm committed:
Wording might sound a little patronising

snopoke committed:
catch generic django not found error

https://sentry.io/dimagi/commcarehq/issues/305082787/

proteusvacuum committed:
Merge pull request #16896 from dimagi/nh/wording

Wording might sound a little patronising

snopoke committed:
Merge pull request #16877 from dimagi/nc/summary-filter

App Summary: Hide case type if no properties match filter

proteusvacuum committed:
Make strings unicode

snopoke committed:
Merge pull request #16897 from dimagi/sk/generic-error

catch generic django not found error

proteusvacuum committed:
Merge pull request #16898 from dimagi/fr/unicode-task

Make strings unicode in eNikshay task

proteusvacuum committed:
Add get_apps_by_id

proteusvacuum committed:
Get all apps if there is no role

proteusvacuum committed:
Only fetch full app if necessary

proteusvacuum committed:
Limit admin view to MAX results

snopoke committed:
Merge pull request #16901 from dimagi/fr/limit-search-results

Limit case search admin view to MAX results

snopoke committed:
Merge pull request #16900 from dimagi/fr/brief-apps-mobile-reports

Speed up mobile reports permission check

kaapstorm committed:
Django UploadedFile has a fileno method

kaapstorm committed:
Leave description as Unicode for now

Commit:
Merge pull request #16887 from dimagi/gc/tag-icds-sms

Add management command to tag older icds sms with indicator metadata

kaapstorm committed:
Merge pull request #16905 from dimagi/nh/uploaded_file

Django's UploadedFile has a fileno method, but you can't use it

Commit:
Update migration

Commit:
Merge branch 'ss/icds-aggregation' of https://github.com/dimagi/commcare-hq into ss/icds-aggregation

Commit:
Remove old migration
